test/hotspot/jtreg/runtime/Nestmates/membership/TestNestHostErrorWithMultiThread.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2021, Huawei Technologies Co., Ltd. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8264760
+ * @summary JVM crashes when two threads encounter the same resolution error
+ *
+ * @library /test/lib
+ * @compile TestNestHostErrorWithMultiThread.java
+ * @compile HostNoNestMember.jcod
+ *
+ * @run main/othervm TestNestHostErrorWithMultiThread
+ */
+
+import java.util.concurrent.CountDownLatch;
+
+class HostNoNestMember {
+  class Member {
+    private int value;
+  }
+
+  public int foo() {
+    Member m = new Member();
+    return m.value;
+  }
+}
+
+public class TestNestHostErrorWithMultiThread {
+
+  public static void main(String args[]) throws Throwable {
+    TestNestHostErrorWithMultiThread t = new TestNestHostErrorWithMultiThread();
+    t.test();
+  }
+
+  public void test() throws Throwable {
+
+    CountDownLatch latch = new CountDownLatch(1);
```

Comment:
The use of CDL doesn't guarantee that both threads have reached the await() before the main thread does the countDown(). If you use a CyclicBarrier instead it will trip when the second thread arrives.

Reply:
I have tested `CyclicBarrier ` here and found that `CountDownLatch` could trigger this bug fast while `CyclicBarrier ` couldn't always trigger this bug.

Reply:
@dholmes-ora

Reply:
> The use of CDL doesn't guarantee that both threads have reached the await() before the main thread does the countDown(). If you use a CyclicBarrier instead it will trip when the second thread arrives.

I have tested a small case  the `CyclicBarrier ` version and `CountDownLatch ` version on `JDK17` & `JDK11`. It seems that :
* For `JDK17`, the wake-up time between two threads in the `CyclicBarrier ` version is bigger than the  `CountDownLatch ` version. In other words,   the  `CountDownLatch ` version can trigger this bug more easily .
* the `CyclicBarrier ` version and   the  `CountDownLatch ` version running under `JDK11`  have the same performance.
```java
// CountDownLatch version 
public class TestCountDownLatch {

  public static void main(String args[]) {
    CountDownLatch latch1 = new CountDownLatch(1);
    CountDownLatch latch2 = new CountDownLatch(2);

    MyThread t1 = new MyThread(latch1, latch2);
    MyThread t2 = new MyThread(latch1, latch2);

    t1.start();
    t2.start();

    try {
      // waiting thread creation
      latch2.await();
      latch1.countDown();

      t1.join();
      t2.join();
    } catch (InterruptedException e) {}   

    System.out.println(Math.abs(t1.getAwakeTime() - t2.getAwakeTime()));
  }

  static class MyThread extends Thread {
    private CountDownLatch latch1;
    private CountDownLatch latch2;
		private long awaketime;

    MyThread(CountDownLatch latch1, CountDownLatch latch2) {
      this.latch1 = latch1;
      this.latch2 = latch2;
    }

    @Override
    public void run() {
      try {
        latch2.countDown();
        // Try to have all threads trigger the nesthost check at the same time
        latch1.await();
        awaketime = System.nanoTime();
      } catch (InterruptedException e) {}
    }

    public long getAwakeTime() {
      return awaketime; 
    }
  }
}
```

```java
// CyclicBarrier version 
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.CyclicBarrier;

public class TestCyclicBarrier {

  public static void main(String args[]) {
    CyclicBarrier barrier = new CyclicBarrier(2);

    MyThread t1 = new MyThread(barrier);
    MyThread t2 = new MyThread(barrier);

    t1.start();
    t2.start();

    try {
      t1.join();
      t2.join();
    } catch (InterruptedException e) {}   

    System.out.println(Math.abs(t1.getAwakeTime() - t2.getAwakeTime()));
  }

  static class MyThread extends Thread {
    private CyclicBarrier barrier; 
		private long awaketime;

    MyThread(CyclicBarrier barrier) {
      this.barrier = barrier;
    }

    @Override
    public void run() {
      try {
        // Try to have all threads trigger the nesthost check at the same time
        barrier.await();
        awaketime = System.nanoTime();
      } catch (Exception e) {}
    }

    public long getAwakeTime() {
      return awaketime; 
    }
  }
}
```
* results:  
 - `CountDownLatch` against `CyclicBarrier`
 ```
  CountDownLatch: 3960
  CyclicBarrier: 3361280
  CountDownLatch: 6120
  CyclicBarrier: 3337540
  CountDownLatch: 6160
  CyclicBarrier: 3462920
  CountDownLatch: 9150
  CyclicBarrier: 3328090
  CountDownLatch: 6580
  CyclicBarrier: 3345450
  CountDownLatch: 6340
  CyclicBarrier: 3342900
  CountDownLatch: 1330
  CyclicBarrier: 3379210
  CountDownLatch: 7780
  CyclicBarrier: 3219020
  CountDownLatch: 2460
  CyclicBarrier: 3297020
  CountDownLatch: 7320
  CyclicBarrier: 3332770
```
-  `JDK17` against `CyclicBarrier`
```
CyclicBarrier jdk17: 3188590
CyclicBarrier jdk11: 49090
CyclicBarrier jdk17: 3123340
CyclicBarrier jdk11: 14680
CyclicBarrier jdk17: 3107910
CyclicBarrier jdk11: 780
CyclicBarrier jdk17: 3072600
CyclicBarrier jdk11: 1720
CyclicBarrier jdk17: 3164340
CyclicBarrier jdk11: 41020
CyclicBarrier jdk17: 3098490
CyclicBarrier jdk11: 7060
CyclicBarrier jdk17: 3058220
CyclicBarrier jdk11: 14750
CyclicBarrier jdk17: 3052460
CyclicBarrier jdk11: 660
CyclicBarrier jdk17: 3083650
CyclicBarrier jdk11: 14670
CyclicBarrier jdk17: 3116260
CyclicBarrier jdk11: 850
```